kedro-datasets/kedro_datasets/_typing.py

```diff
@@ -6,9 +6,9 @@
 from typing import NewType
 
 TablePreview = NewType("TablePreview", dict)
-ImagePreview = NewType("ImagePreview", bytes)
+ImagePreview = NewType("ImagePreview", str)
 PlotlyPreview = NewType("PlotlyPreview", dict)
-JSONPreview = NewType("JSONPreview", dict)
+JSONPreview = NewType("JSONPreview", str)
 
```

Comment:
@SajidAlamQB @rashidakanchwala please double check this

Reply:
It's a bit confusing because `tracking.JSONDataset` returns `json.loads()`, which is a `dict`, while `json.JSONDataset` returns `json.dumps()`, resulting in a `string`.

Reply:
`tracking.JSONDataset.preview()` returns `JSONTrackingPreview` type which is still `dict`
`json.JSONDataset.preview()` returns `JSONPreview` which I've changed to `str`

Since `tracking.JSONDataset` is a subclass of `json.JSONDataset` `mypy` does complain about the overwriting of methods but I've added an ignore - is that okay?